### PR TITLE
RGB Matrix working for Sofle RGB

### PR DESCRIPTION
**Pull request by killmaster**
Sofle RGB working with RGB Matrix and with both halves properly synced

## Description

Added the `g_led_config` to the rev1.c file (based it off of Djinn's rev2 and crkbdr rev1).
Completed the `config.h` file on the `killmaster` keymap to work with the RGB Matrix.
Added more functionality to the encoders, so they can control the hue, saturation, value and speed of the matrix.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sofle/keymaps/killmaster/encoder.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            uint16_t mapped_code        = 0;
+            if (clockwise) {
+                mapped_code = KC_VOLD;
+            } else {
+                mapped_code = KC_VOLU;
+            }
+            register_code(mapped_code);
+            while (timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY)
+                ; /* no-op */
+            unregister_code(mapped_code);
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/sofle/keymaps/killmaster/encoder.c:70]: Also, why not use `tap_code` and `#define TAP_CODE_DELAY 2`?

File: keyboards/sofle/rev1/rev1.c [hunk 2]
```diff
@@ -1 +1,81 @@
 #include "sofle.h"
```
Comment (drashna) [inline keyboards/sofle/rev1/rev1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/sofle/rev1/rev1.c [hunk 3]
```diff
@@ -1 +1,81 @@
...
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT,
+        LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT, LED_FLAG_KEYLIGHT
+    }
+};
+#endif
+
+void matrix_init_kb(void) {
+    matrix_init_user();
+    //rgb_matrix_mode(RGB_MATRIX_CYCLE_LEFT_RIGHT);
+}
```
Comment (drashna) [inline keyboards/sofle/rev1/rev1.c:81]:
```suggestion
```

File: keyboards/sofle/rev1/rev1.c [hunk 4]
```diff
@@ -1 +1,81 @@
...
+  // 10 15 20 25 30 35       35 30 25 20 15 10  2
+  //
+  // 09 16 19 26 29 36       36 29 26 19 16 09  3
+  //
+  //     08 17 18 27 28     28 27 18 17 08      4
+  //    07    06    05       05    06    07
+
+led_config_t g_led_config = {
+    {
+        {  LLI(11),  LLI(12),  LLI(21),  LLI(22),  LLI(31),  LLI(32) },
```
Comment (drashna) [inline keyboards/sofle/rev1/rev1.c:26]: Please change this to the normal format for g_led_config.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sofle/keymaps/killmaster/encoder.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (killmaster) [reply keyboards/sofle/keymaps/killmaster/encoder.c:70]: Didn't occur to me. Got this from the Djinn's implementation and didn't pay much attention.
Going to try it out.

File: keyboards/sofle/rev1/rev1.c (hunk 4 above)
Comment (killmaster) [reply keyboards/sofle/rev1/rev1.c:26]: Is that by removing the macros?

Comment (drashna) [reply keyboards/sofle/rev1/rev1.c:26]: Yeah.

File: keyboards/sofle/keymaps/killmaster/encoder.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/sofle/keymaps/killmaster/encoder.c:70]: no worries.  And if you want a different value for the deal for some of the tap_code functions, `tap_code_delay` allows a custom wait setting (eg `tap_code(kc, ms)`)

File: keyboards/sofle/rev1/rev1.c (hunk 4 above)
Comment (killmaster) [reply keyboards/sofle/rev1/rev1.c]: Is it okay like this? Should I also remove the comments?

File: keyboards/sofle/keymaps/killmaster/rules.mk [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+EXTRAKEY_ENABLE = yes
+VIA_ENABLE = no
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+RGBLIGHT_DRIVER = WS2812
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+SPLIT_KEYBOARD = yes
+MOUSEKEY_ENABLE = yes
+SPLIT_TRANSPORT_MIRROR = yes
```
Comment (drashna) [inline keyboards/sofle/keymaps/killmaster/rules.mk:14]:
```suggestion
```

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/sofle/rev1/readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: SofleKeyboard PCB, ProMicro  
+* Hardware Availability: [PCB & Case Data](https://github.com/josefadamcik/SofleKeyboard)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make sofle:default
+
+Flash the default keymap: 
+
+    make sofle:default:avrdude
```
Comment (fauxpark) [inline keyboards/sofle/rev1/readme.md:21]:
```suggestion
Flashing example for this keyboard:

    make sofle:default:flash
```

File: keyboards/sofle/keymaps/killmaster/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes
+VIA_ENABLE = no
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+RGBLIGHT_DRIVER = WS2812
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+SPLIT_KEYBOARD = yes
+MOUSEKEY_ENABLE = no
+POINTING_DEVICE_ENABLE = no
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/killmaster/rules.mk:21]:
```suggestion
EXTRAKEY_ENABLE = yes
LTO_ENABLE = yes
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
```

File: keyboards/sofle/keymaps/killmaster/keymap.c [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+
+[_QWERTY] = LAYOUT( \
+  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV, \
+  KC_ESC,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,  KC_BSPC, \
+  KC_TAB,   KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN,  KC_QUOT, \
+  KC_LSPO,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,     KC_MPLY,KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSPC, \
+                 KC_LGUI,KC_LALT,KC_LCTRL, KC_LOWER, KC_SPC,      KC_ENT,  KC_RAISE, KC_RCTRL, KC_RALT, KC_RGUI \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/killmaster/keymap.c:63]:
```suggestion
[_QWERTY] = LAYOUT(
  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV,
  KC_ESC,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,  KC_BSPC,
  KC_TAB,   KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN,  KC_QUOT,
  KC_LSPO,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,     KC_MPLY,KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSPC,
                 KC_LGUI,KC_LALT,KC_LCTRL, KC_LOWER, KC_SPC,      KC_ENT,  KC_RAISE, KC_RCTRL, KC_RALT, KC_RGUI
```

File: keyboards/sofle/keymaps/killmaster/keymap.c [hunk 9]
```diff
@@ -0,0 +1,402 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[_LOWER] = LAYOUT( \
+  _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                       KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,\
+  KC_GRV,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                       KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_F12, \
+  _______, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC,                       KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE, \
+  _______,  KC_EQL, KC_MINS, KC_PLUS, KC_LCBR, KC_RCBR, _______,       _______, KC_LBRC, KC_RBRC, KC_SCLN, KC_COLN, KC_BSLS, _______, \
+                       _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______\
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/killmaster/keymap.c:84]:
```suggestion
[_LOWER] = LAYOUT(
  _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                       KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,
  KC_GRV,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                       KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_F12,
  _______, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC,                       KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE,
  _______,  KC_EQL, KC_MINS, KC_PLUS, KC_LCBR, KC_RCBR, _______,       _______, KC_LBRC, KC_RBRC, KC_SCLN, KC_COLN, KC_BSLS, _______,
                       _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______
```

File: keyboards/sofle/keymaps/killmaster/keymap.c [hunk 10]
```diff
@@ -0,0 +1,402 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[_RAISE] = LAYOUT( \
+  _______, _______ , _______ , _______ , RGB_RMOD , RGB_MOD,                           _______,  _______  , _______,  _______ ,  _______ ,_______, \
+  _______,  KC_INS,  KC_PSCR,   KC_APP,  XXXXXXX, RGB_TOG,                        KC_PGUP, KC_PRVWD,   KC_UP, KC_NXTWD,KC_DLINE, KC_BSPC, \
+  _______, KC_LALT,  KC_LCTL,  KC_LSFT,  XXXXXXX, KC_CAPS,                       KC_PGDN,  KC_LEFT, KC_DOWN, KC_RGHT,  KC_DEL, KC_BSPC, \
+  _______,XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,  _______,       _______,  XXXXXXX, KC_LSTRT, XXXXXXX, KC_LEND,   XXXXXXX, _______, \
+                         _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/killmaster/keymap.c:105]:
```suggestion
[_RAISE] = LAYOUT(
  _______, _______ , _______ , _______ , RGB_RMOD , RGB_MOD,                           _______,  _______  , _______,  _______ ,  _______ ,_______,
  _______,  KC_INS,  KC_PSCR,   KC_APP,  XXXXXXX, RGB_TOG,                        KC_PGUP, KC_PRVWD,   KC_UP, KC_NXTWD,KC_DLINE, KC_BSPC,
  _______, KC_LALT,  KC_LCTL,  KC_LSFT,  XXXXXXX, KC_CAPS,                       KC_PGDN,  KC_LEFT, KC_DOWN, KC_RGHT,  KC_DEL, KC_BSPC,
  _______,XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,  _______,       _______,  XXXXXXX, KC_LSTRT, XXXXXXX, KC_LEND,   XXXXXXX, _______,
                         _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______
```

File: keyboards/sofle/keymaps/killmaster/keymap.c [hunk 11]
```diff
@@ -0,0 +1,402 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+  [_ADJUST] = LAYOUT( \
+  XXXXXXX , XXXXXXX,  XXXXXXX ,  XXXXXXX , XXXXXXX, XXXXXXX,                     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  RESET  , XXXXXXX, KC_QWERTY,  XXXXXXX , CG_TOGG,XXXXXXX,                     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  XXXXXXX , XXXXXXX,CG_TOGG, XXXXXXX,    XXXXXXX,  XXXXXXX,                     XXXXXXX, KC_VOLD, KC_MUTE, KC_VOLU, XXXXXXX, XXXXXXX, \
+  XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX,    XXXXXXX,  XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX, KC_MPRV, KC_MPLY, KC_MNXT, XXXXXXX, XXXXXXX, \
+                   _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/killmaster/keymap.c:126]:
```suggestion
  [_ADJUST] = LAYOUT(
  XXXXXXX , XXXXXXX,  XXXXXXX ,  XXXXXXX , XXXXXXX, XXXXXXX,                     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
  RESET  , XXXXXXX, KC_QWERTY,  XXXXXXX , CG_TOGG,XXXXXXX,                     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
  XXXXXXX , XXXXXXX,CG_TOGG, XXXXXXX,    XXXXXXX,  XXXXXXX,                     XXXXXXX, KC_VOLD, KC_MUTE, KC_VOLU, XXXXXXX, XXXXXXX,
  XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX,    XXXXXXX,  XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX, KC_MPRV, KC_MPLY, KC_MNXT, XXXXXXX, XXXXXXX,
                   _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sofle/keymaps/killmaster/rules.mk (hunk 7 above)
Comment (killmaster) [reply keyboards/sofle/keymaps/killmaster/rules.mk]: Cleaned up the file and tested it.
Worked perfectly.
Thank you for the suggestion!

**Review by fauxpark — APPROVED** [review]

**Pull request merged**